docs/config/build-time.adoc

```diff
@@ -0,0 +1,228 @@
+= React4XP - Configuration - Build time
+:toc: right
+
+[[react4xp.config.js]]
+== react4xp.config.js
+
+Several aspects of React4xp can be configured with a file _react4xp.config.js_.
+
+If you're using the <<setup#create_project, starter>> or your project is based on it, this is already set up and you can skip the following section:
+
+{zwsp} +
+
+=== Including react4xp.config.js in your project
+
+Put _react4xp.config.js_ at the root of your XP project. A template can be copied from link:https://github.com/enonic/enonic-react4xp/blob/master/examples/react4xp.config.js[here], or from _node_modules/@enonic/react4xp/examples/react4xp.config.js_ after installing the React4xp NPM package.
+
+*In order to take effect, it must be handled by gradle during the build:*
+
+- Add this to your _build.gradle_:
++
+[source,groovy,options="nowrap"]
+----
+task react4xp(type: NpmTask, dependsOn: npmInstall) {
+  args = [
+    'run',
+    'build:react4xp' // This script must exist in the package.json file
+  ]
+  description 'Compile react4xp resources'
+  environment = [
+    'R4X_APP_NAME': "${appName}",
+    'R4X_BUILD_LOG_LEVEL': gradle.startParameter.logLevel.toString(),
+    'R4X_DIR_PATH_ABSOLUTE_PROJECT': project.projectDir.toString(),
+    'NODE_ENV': project.hasProperty('dev') || project.hasProperty('development') ? 'development' : 'production'
+  ]
+  group 'react4xp'
+  // It also watches package.json and package-lock.json :)
+  inputs.dir 'node_modules/@enonic/react4xp'
+  inputs.dir 'src/main/resources'
+  outputs.dir 'build/resources/main'
```

Comment:
```suggestion
    inputs.files fileTree(dir: 'src/main/resources/site', include: ['**/*.tsx', '**/*.jsx'])
    inputs.files('package.json', 'package-lock.json', 'webpack.config.react4xp.js', 'react4xp.config.js', 'tsconfig.react4xp.json')
    inputs.dir 'src/main/resources/react4xp'
    outputs.dir 'build/resources/main/assets/react4xp'
```

Reply:
There can be tsx and jsx files outside src/main/resources/site. In fact there can even be ts, js, es, es6 and so on.

Reply:
https://github.com/enonic/enonic-react4xp/blob/master/examples/react4xp.config.js#L72

Reply:
sure, but in this case people should add such inputs themselves

Reply:
The default extenstions are 'jsx', 'tsx', 'ts', 'es6', 'es', 'js' 
https://github.com/enonic/enonic-react4xp/blob/c5d7855925f7e24c26c1edd1af3265c0474dcbb1/src/webpack.config.components.ts#L77

Reply:
We can, and it works, here is an example:
https://github.com/enonic/starter-react4xp/blob/master/src/main/resources/site/parts/example/example.ts#L2